pkg/controller/deadmanssnitchintegration/deadmanssnitchintegration_controller_test.go

```diff
@@ -307,6 +323,50 @@ func TestReconcileClusterDeployment(t *testing.T) {
 				r.Delete(gomock.Any()).Times(0)
 			},
 		},
+		{
+			name: "Test Managed noalerts ClusterDeployment",
+			localObjects: []runtime.Object{
+				testSecret(),
+				noalertsManagedClusterDeployment(),
+				testDeadMansSnitchIntegration(),
+			},
+			expectedSyncSets: &SyncSetEntry{},
+			expectedSecret:   &SecretEntry{},
+			verifySyncSets:   verifyNoSyncSet,
+			verifySecret:     verifyNoSecret,
+			setupDMSMock: func(r *mockdms.MockClientMockRecorder) {
+				r.Create(gomock.Any()).Times(0)
```

Comment:
I am confused by this test, I would expect it to try to setup DMS since the "noalerts" labels have no meaning anymore in the code and the test DMSI only matches on managed=true.

Reply:
Yeah, you're totally correct here. This test was only working because it was using a non-managed clusterdeployment with no DMS finalizer, and alerts didn't factor into it at all. Thanks for that catch.

In the branch now, I have created a new test as a replacement - one that will return a non-managed cluster with a DMS finalizer in it. This will exercise the new behaviour that we want this controller to take - for a cluster that doesn't match the `clusterDeploymentSelector` but has a DMS finalizer  -> delete the DMS setup. I think this is what we actually want here.

Thanks a tonne for noticing this - very much a :facepalm: moment!

Reply:
Fwiw I spent time in the PD operator to refactor tests and was considering doing it for DMS.  Not asking for that change here but may be something to consider in the future.